jfix-stdlib-ratelimiter/src/main/kotlin/ru/fix/stdlib/ratelimiter/SwitchableRateLimitedDispatcher.kt

```diff
@@ -0,0 +1,88 @@
+package ru.fix.stdlib.ratelimiter
+
+import ru.fix.aggregating.profiler.Profiler
+import ru.fix.dynamic.property.api.DynamicProperty
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+import java.util.function.Supplier
+import kotlin.coroutines.CoroutineContext
+
+class SwitchableRateLimitedDispatcher(
+    private val name: String,
+    private val rateLimiterKt: RateLimiterKt,
+    private val rateLimiter: RateLimiter,
+    private val profiler: Profiler,
+    private val context: CoroutineContext,
+    private val rateLimiterSettings: DynamicProperty<RateLimiterSettings>,
+) : RateLimitedDispatcherInterface {
+
+    companion object {
+        private const val AWAIT_MS = 1_000L
+    }
+
+    private var rateLimitedDispatcher: AtomicReference<RateLimitedDispatcherInterface?> = AtomicReference(null)
+    private var isSuspendable = AtomicReference(rateLimiterSettings.get().isSuspendable)
+    private var countDownLatch = CountDownLatch(1)
+
+    init {
+        rateLimiterSettings.map { it.isSuspendable }
+            .createSubscription()
+            .setAndCallListener { _, newValue ->
+                if (isSuspendable.get() != newValue) {
+                    updateDispatcher(newValue)
+                }
+            }
+        updateDispatcher(isSuspendable.get())
+    }
+
+    override fun <T> compose(supplier: Supplier<CompletableFuture<T>>): CompletableFuture<T> {
```

Comment:
is it ok that we do not delegate decorateSuspend?

Reply:
Added override